### ISSUE TITLE
Can't build the manager in Arch Linux

### ISSUE THREAD

**Issue by BraulioV** [wazuh/wazuh#3012]
|Wazuh version|Install type|Install method|Platform|
|---|---|---|---|
| 3.9.0 | Manager | Sources | Rolling release |

Hi team, 

if you try to build the manager it will fail linking when linking the binaries  with the `libwazuhext.so`. Here you can see the output:

```Makefile
   CC wazuh-modulesd
/usr/bin/ld: libwazuhext.so: undefined reference to `nghttp2_session_set_stream_user_data'
/usr/bin/ld: libwazuhext.so: undefined reference to `nghttp2_session_upgrade'
/usr/bin/ld: libwazuhext.so: undefined reference to `BrotliDecoderDestroyInstance'
/usr/bin/ld: libwazuhext.so: undefined reference to `nghttp2_session_callbacks_set_on_frame_send_callback'
/usr/bin/ld: libwazuhext.so: undefined reference to `nghttp2_submit_rst_stream'
/usr/bin/ld: libwazuhext.so: undefined reference to `nghttp2_session_callbacks_set_on_stream_close_callback'
/usr/bin/ld:/usr libwazuhext.so/bin/:ld:  libwazuhext.so:undefined  reference undefinedto  `nghttp2_session_get_remote_settingsreference'
 /usrto `nghttp2_session_set_stream_user_data'
/usr/bin/ld: libwazuhext.so: undefined reference to `nghttp2_session_upgrade/bin'/ld
/: usr/bin/ld: libwazuhext.so: undefinedlibwazuhext.so: reference to `BrotliDecoderDestroyInstance undefined'
/usr/bin reference/ld: to  libwazuhext.so:`RTMP_Init undefined reference to'
 /`usrnghttp2_session_callbacks_set_on_frame_send_callback/bin'
/usr/bin/ld: libwazuhext.so: undefined reference to `nghttp2_submit_rst_stream'
/usr/bin/ld: libwazuhext.so: undefined reference to `nghttp2_session_callbacks_set_on_stream_close_callback'
/usr/bin/ld: libwazuhext.so: undefined reference to `nghttp2_session_get_remote_settings'
/usr/bin/ld: libwazuhext.so: undefined reference to `RTMP_Init'
/usr/bin/ld: libwazuhext.so: undefined reference to `nghttp2_priority_spec_init'
/usr/bin/ld: libwazuhext.so: undefined reference to `RTMP_SetBufferMS'
/usr/bin/ld: libwazuhext.so: undefined reference to `nghttp2_session_callbacks_set_error_callback'
/usr/bin/ld: libwazuhext.so: undefined reference to `nghttp2_session_callbacks_new'
/usr/bin/ld: libwazuhext.so: undefined reference to `nghttp2_submit_priority/ld: libwazuhext.so: undefined reference to `nghttp2_priority_spec_init'
/usr/bin/ld: libwazuhext.so: undefined reference to `RTMP_SetBufferMS'
/usr/bin/ld: libwazuhext.so: undefined reference to `nghttp2_session_callbacks_set_error_callback'
/usr/bin/ld: libwazuhext.so: undefined'
 reference/usr to/bin `nghttp2_session_callbacks_new'
/usr/bin/ld: libwazuhext.so: undefined reference to `nghttp2_submit_priority'
/usr/bin/ld: libwazuhext.so/ld: libwazuhext.so: undefined reference to `nghttp2_session_mem_recv'
/usr/bin/ld: libwazuhext.so: undefined: undefined reference to `nghttp2_session_mem_recv'
/usr/bin/ld: libwazuhext.so: undefined reference to `nghttp2_version'
/usr/bin/ld: libwazuhext.so reference to `nghttp2_version'
/usr/bin/ld: libwazuhext.so: undefined reference to `nghttp2_session_callbacks_set_on_invalid_frame_recv_callback'
/usr/bin/: ld:undefined  libwazuhext.soreference : to undefined `nghttp2_session_callbacks_set_on_invalid_frame_recv_callbackreference '
to /usr`RTMP_ConnectStream/bin'
/ld/usr: /binlibwazuhext.so:/ld undefined:  referencelibwazuhext.so: to undefined ` referenceRTMP_ConnectStream' to
/ `usr/nghttp2_session_set_local_window_size'bin/
/ld:usr/ libwazuhext.sobin/: ld:undefined  libwazuhext.soreference : to undefined `nghttp2_session_set_local_window_sizereference '
to `/usrnghttp2_http2_strerror'/bin
//ldusr/: bin/libwazuhext.so:ld: undefined libwazuhext.so reference:  toundefined  `reference nghttp2_http2_strerror'to 
/`BrotliDecoderCreateInstanceusr/'
bin//usrld:/bin libwazuhext.so/ld: : undefined libwazuhext.so:reference  undefinedto  reference`BrotliDecoderCreateInstance to'
 `/usrnghttp2_is_fatal'/bin
//ldusr/: bin/libwazuhext.so:ld: undefined libwazuhext.so reference:  toundefined  `reference nghttp2_is_fatal'to 
/`RTMP_Allocusr/bin/ld: libwazuhext.so:'
 undefined/usr reference/bin to/ld `: RTMP_Alloc'libwazuhext.so:
/ undefinedusr/ referencebin/ told: ` libwazuhext.songhttp2_session_resume_data': 
/undefined usr/reference bin/to ld:`nghttp2_session_resume_data libwazuhext.so'
: /usr/bin/ld: libwazuhext.soundefined reference to `nghttp2_session_client_new: '
undefined /usrreference /binto /ld`nghttp2_session_client_new: '
libwazuhext.so:/usr    CC wazuh-db
 undefined reference /bin/to ld:`psl_builtin libwazuhext.so'
: /usrundefined /binreference /ldto : libwazuhext.so: undefined reference to `nghttp2_submit_request`psl_builtin'
/usr/bin/ld: libwazuhext.so: undefined reference to `nghttp2_submit_request'
/usr/bin/ld: libwazuhext.so: undefined reference to `nghttp2_session_callbacks_set_on_header_callback'
/usr/bin/ld: libwazuhext.so: undefined reference to `nghttp2_session_want_write'
/usr/bin/ld: libwazuhext.so: undefined reference to `RTMP_Read'
/usr/bin/ld: libwazuhext.so: undefined reference to `RTMP_SetupURL'
/usr/bin/ld': 
/usr/bin/ld: libwazuhext.so: undefined reference to `nghttp2_session_callbacks_set_on_header_callback'
/usr/bin/ld: libwazuhext.so: undefined reference to `nghttp2_session_want_write'
/usr/bin/ld: libwazuhext.so: undefined reference to `RTMP_Read'
/usr/bin/ld: libwazuhext.so: undefined reference to `RTMP_SetupURL'
/usr/bin/ldlibwazuhext.so:: libwazuhext.so: undefined reference to `BrotliDecoderGetErrorCode'
/usr/bin/ld: libwazuhext.so: undefined reference to `nghttp2_session_want_read undefined'
/usr/bin/ld: libwazuhext.so: undefined reference to `nghttp2_session_del'
/usr/bin/ld: libwazuhext.so: undefined reference to `RTMP_Free'
/usr/bin/ld: libwazuhext.so: undefined reference reference to `nghttp2_pack_settings_payload'
/usr/bin/ld: libwazuhext.so: undefined reference to `nghttp2_strerror'
/usr/bin/ld: libwazuhext.so: undefined reference to `nghttp2_session_get_stream_user_data'
/usr/bin/ld: libwazuhext.so: undefined reference to `BrotliDecoderDecompressStream'
/usr/bin/ld to: libwazuhext.so: undefined reference to `psl_is_cookie_domain_acceptable'
/usr/bin/ld: libwazuhext.so: undefined reference to `nghttp2_session_callbacks_set_send_callback'
/usr/bin/ld: libwazuhext.so: undefined reference to `nghttp2_session_callbacks_set_before_frame_send_callback'
/usr/bin/ld: libwazuhext.so: undefined reference to `RTMP_Write'
/usr/bin/ld: libwazuhext.so: undefined reference to `nghttp2_session_callbacks_del'
/usr/bin/ld: libwazuhext.so: undefined reference to `nghttp2_session_callbacks_set_on_begin_headers_callback'
/usr/bin/ld: libwazuhext.so: undefined reference to `nghttp2_session_callbacks_set_on_frame_recv_callback'
/usr/bin/ld: libwazuhext.so: undefined reference to `psl_get_version'
/usr/bin/ld: libwazuhext.so: undefined reference to `RTMP_Close'
/usr/bin/ld: libwazuhext.so: undefined reference to `nghttp2_session_send'
/usr/bin/ld: libwazuhext.so: undefined reference to `RTMP_Connect1'
/usr/bin/ld: libwazuhext.so: undefined reference to ` `nghttp2_session_callbacks_set_on_data_chunk_recv_callback'
/usr/bin/ld: libwazuhext.so: undefined reference to `nghttp2_session_callbacks_set_on_frame_not_send_callback'
/usr/binBrotliDecoderGetErrorCode'/ld:
/ libwazuhext.so: undefined reference to `BrotliDecoderVersion'
/usr/bin/ld: libwazuhext.so: undefined reference to `nghttp2_submit_settings'
usr/bin/ld: libwazuhext.so: undefined referencecollect2: error: ld returned 1 exit status
 to `nghttp2_session_want_read'
/usr/bin/ld: libwazuhext.so: undefined reference to `nghttp2_session_del'
/usr/bin/ld: libwazuhext.so: undefined reference to `RTMP_Free'
/usr/bin/ld: libwazuhext.so: undefined reference to `nghttp2_pack_settings_payload'
/usr/bin/ld: libwazuhext.so: undefined reference to `nghttp2_strerror'
/usr/bin/ld: libwazuhext.so: undefined reference to `nghttp2_session_get_stream_user_data'
/usr/bin/ld: libwazuhext.so: undefined reference to `BrotliDecoderDecompressStream'
/usr/bin/ld: libwazuhext.so: undefined reference to `psl_is_cookie_domain_acceptable'
/usr/bin/ld: libwazuhext.so: undefined reference to `nghttp2_session_callbacks_set_send_callback'
/usr/bin/ld: libwazuhext.so: undefined reference to `nghttp2_session_callbacks_set_before_frame_send_callback'
/usr/bin/ld: libwazuhext.so: undefined reference to `RTMP_Write'
/usr/bin/ld: libwazuhext.so: undefined reference to `nghttp2_session_callbacks_del'
/usr/bin/ld: libwazuhext.so: undefined reference to `nghttp2_session_callbacks_set_on_begin_headers_callback'
/usr/bin/ld: libwazuhext.so: undefined reference to `nghttp2_session_callbacks_set_on_frame_recv_callback'
/usr/bin/ldmake[1]: *** [Makefile:1024: ossec-dbd] Error 1
: libwazuhext.so: undefined reference make[1]: *** Waiting for unfinished jobs....
to `psl_get_version'
/usr/bin/ld: libwazuhext.so: undefined reference to `RTMP_Close'
/usr/bin/ld: libwazuhext.so: undefined reference to `nghttp2_session_send'
/usr/bin/ld: libwazuhext.so: undefined reference to `RTMP_Connect1'
/usr/bin/ld: libwazuhext.so: undefined reference to `nghttp2_session_callbacks_set_on_data_chunk_recv_callback'
/usr/bin/ld: libwazuhext.so: undefined reference to `nghttp2_session_callbacks_set_on_frame_not_send_callback'
/usr/bin/ld: libwazuhext.so: undefined reference to `BrotliDecoderVersion'
/usr/bin/ld: libwazuhext.so: undefined reference to `nghttp2_submit_settings'
collect2: error: ld returned 1 exit status
make[1]: *** [Makefile:1239: ossec-integratord] Error 1
/usr/bin/ld: libwazuhext.so: undefined reference to `nghttp2_session_set_stream_user_data'
/usr/bin/ld: libwazuhext.so: undefined reference to `nghttp2_session_upgrade'
/usr/bin/ld: libwazuhext.so: undefined reference to `BrotliDecoderDestroyInstance'
/usr/bin/ld: libwazuhext.so: undefined reference to `nghttp2_session_callbacks_set_on_frame_send_callback'
/usr/bin/ld: libwazuhext.so: undefined reference to `nghttp2_submit_rst_stream'
/usr/bin/ld: libwazuhext.so: undefined reference to `nghttp2_session_callbacks_set_on_stream_close_callback'
/usr/bin/ld: libwazuhext.so: undefined reference to `nghttp2_session_get_remote_settings'
/usr/bin/ld: libwazuhext.so: undefined reference to `RTMP_Init'
/usr/bin/ld: libwazuhext.so: undefined reference to `nghttp2_priority_spec_init'
/usr/bin/ld: libwazuhext.so: undefined reference to `RTMP_SetBufferMS'
/usr/bin/ld: libwazuhext.so: undefined reference to `nghttp2_session_callbacks_set_error_callback'
/usr/bin/ld: libwazuhext.so: undefined reference to `nghttp2_session_callbacks_new'
/usr/bin/ld: libwazuhext.so: undefined reference to `nghttp2_submit_priority'
/usr/bin/ld: libwazuhext.so: undefined reference to `nghttp2_session_mem_recv'
/usr/bin/ld: libwazuhext.so: undefined reference to `nghttp2_version'
/usr/bin/ld: libwazuhext.so: undefined reference to `nghttp2_session_callbacks_set_on_invalid_frame_recv_callback'
/usr/bin/ld: libwazuhext.so: undefined reference to `RTMP_ConnectStream'
/usr/bin/ld: libwazuhext.so: undefined reference to `nghttp2_session_set_local_window_size'
/usr/bin/ld: libwazuhext.so: undefined reference to `nghttp2_http2_strerror'
/usr/bin/ld: libwazuhext.so: undefined reference to `BrotliDecoderCreateInstance'
/usr/bin/ld: libwazuhext.so: undefined reference to `nghttp2_is_fatal'
/usr/bin/ld: libwazuhext.so: undefined reference to `RTMP_Alloc'
/usr/bin/ld: libwazuhext.so: undefined reference to `nghttp2_session_resume_data'
/usr/bin/ld: libwazuhext.so: undefined reference to `nghttp2_session_client_new'
/usr/bin/ld: libwazuhext.so: undefined reference to `psl_builtin'
/usr/bin/ld: libwazuhext.so: undefined reference to `nghttp2_submit_request'
/usr/bin/ld: libwazuhext.so: undefined reference to `nghttp2_session_callbacks_set_on_header_callback'
/usr/bin/ld: libwazuhext.so: undefined reference to `nghttp2_session_want_write'
/usr/bin/ld: libwazuhext.so: undefined reference to `RTMP_Read'
/usr/bin/ld: libwazuhext.so: undefined reference to `RTMP_SetupURL'
/usr/bin/ld: libwazuhext.so: undefined reference to `BrotliDecoderGetErrorCode'
/usr/bin/ld: libwazuhext.so: undefined reference to `nghttp2_session_want_read'
/usr/bin/ld: libwazuhext.so: undefined reference to `nghttp2_session_del'
/usr/bin/ld: libwazuhext.so: undefined reference to `RTMP_Free'
/usr/bin/ld: libwazuhext.so: undefined reference to `nghttp2_pack_settings_payload'
/usr/bin/ld: libwazuhext.so: undefined reference to `nghttp2_strerror'
/usr/bin/ld: libwazuhext.so: undefined reference to `nghttp2_session_get_stream_user_data'
/usr/bin/ld: libwazuhext.so: undefined reference to `BrotliDecoderDecompressStream'
/usr/bin/ld: libwazuhext.so: undefined reference to `psl_is_cookie_domain_acceptable'
/usr/bin/ld: libwazuhext.so: undefined reference to `nghttp2_session_callbacks_set_send_callback'
/usr/bin/ld: libwazuhext.so: undefined reference to `nghttp2_session_callbacks_set_before_frame_send_callback'
/usr/bin/ld: libwazuhext.so: undefined reference to `RTMP_Write'
/usr/bin/ld: libwazuhext.so: undefined reference to `nghttp2_session_callbacks_del'
/usr/bin/ld: libwazuhext.so: undefined reference to `nghttp2_session_callbacks_set_on_begin_headers_callback'
/usr/bin/ld: libwazuhext.so: undefined reference to `nghttp2_session_callbacks_set_on_frame_recv_callback'
/usr/bin/ld: libwazuhext.so: undefined reference to `psl_get_version'
/usr/bin/ld: libwazuhext.so: undefined reference to `RTMP_Close'
/usr/bin/ld: libwazuhext.so: undefined reference to `nghttp2_session_send'
/usr/bin/ld: libwazuhext.so: undefined reference to `RTMP_Connect1'
/usr/bin/ld: libwazuhext.so: undefined reference to `nghttp2_session_callbacks_set_on_data_chunk_recv_callback'
/usr/bin/ld: libwazuhext.so: undefined reference to `nghttp2_session_callbacks_set_on_frame_not_send_callback'
/usr/bin/ld: libwazuhext.so: undefined reference to `BrotliDecoderVersion'
/usr/bin/ld: libwazuhext.so: undefined reference to `nghttp2_submit_settings'
collect2: error: ld returned 1 exit status
make[1]: *** [Makefile:1352: wazuh-modulesd] Error 1
/usr/bin/ld: wazuh_db/wdb_fim.o: in function `wdb_syscheck_load':
wdb_fim.c:(.text+0xd05): undefined reference to `sk_build_sum'
/usr/bin/ld: wazuh_db/wdb_fim.o: in function `wdb_syscheck_save':
wdb_fim.c:(.text+0x13c3): undefined reference to `sk_decode_extradata'
/usr/bin/ld: wdb_fim.c:(.text+0x13d8): undefined reference to `sk_decode_sum'
/usr/bin/ld: wdb_fim.c:(.text+0x1429): undefined reference to `sk_sum_clean'
/usr/bin/ld: libwazuhext.so: undefined reference to `nghttp2_session_set_stream_user_data'
/usr/bin/ld: libwazuhext.so: undefined reference to `nghttp2_session_upgrade'
/usr/bin/ld: libwazuhext.so: undefined reference to `BrotliDecoderDestroyInstance'
/usr/bin/ld: libwazuhext.so: undefined reference to `nghttp2_session_callbacks_set_on_frame_send_callback'
/usr/bin/ld: libwazuhext.so: undefined reference to `nghttp2_submit_rst_stream'
/usr/bin/ld: libwazuhext.so: undefined reference to `nghttp2_session_callbacks_set_on_stream_close_callback'
/usr/bin/ld: libwazuhext.so: undefined reference to `nghttp2_session_get_remote_settings'
/usr/bin/ld: libwazuhext.so: undefined reference to `RTMP_Init'
/usr/bin/ld: libwazuhext.so: undefined reference to `nghttp2_priority_spec_init'
/usr/bin/ld: libwazuhext.so: undefined reference to `RTMP_SetBufferMS'
/usr/bin/ld: libwazuhext.so: undefined reference to `nghttp2_session_callbacks_set_error_callback'
/usr/bin/ld: libwazuhext.so: undefined reference to `nghttp2_session_callbacks_new'
/usr/bin/ld: libwazuhext.so: undefined reference to `nghttp2_submit_priority'
/usr/bin/ld: libwazuhext.so: undefined reference to `nghttp2_session_mem_recv'
/usr/bin/ld: libwazuhext.so: undefined reference to `nghttp2_version'
/usr/bin/ld: libwazuhext.so: undefined reference to `nghttp2_session_callbacks_set_on_invalid_frame_recv_callback'
/usr/bin/ld: libwazuhext.so: undefined reference to `RTMP_ConnectStream'
/usr/bin/ld: libwazuhext.so: undefined reference to `nghttp2_session_set_local_window_size'
/usr/bin/ld: libwazuhext.so: undefined reference to `nghttp2_http2_strerror'
/usr/bin/ld: libwazuhext.so: undefined reference to `BrotliDecoderCreateInstance'
/usr/bin/ld: libwazuhext.so: undefined reference to `nghttp2_is_fatal'
/usr/bin/ld: libwazuhext.so: undefined reference to `RTMP_Alloc'
/usr/bin/ld: libwazuhext.so: undefined reference to `nghttp2_session_resume_data'
/usr/bin/ld: libwazuhext.so: undefined reference to `nghttp2_session_client_new'
/usr/bin/ld: libwazuhext.so: undefined reference to `psl_builtin'
/usr/bin/ld: libwazuhext.so: undefined reference to `nghttp2_submit_request'
/usr/bin/ld: libwazuhext.so: undefined reference to `nghttp2_session_callbacks_set_on_header_callback'
/usr/bin/ld: libwazuhext.so: undefined reference to `nghttp2_session_want_write'
/usr/bin/ld: libwazuhext.so: undefined reference to `RTMP_Read'
/usr/bin/ld: libwazuhext.so: undefined reference to `RTMP_SetupURL'
/usr/bin/ld: libwazuhext.so: undefined reference to `BrotliDecoderGetErrorCode'
/usr/bin/ld: libwazuhext.so: undefined reference to `nghttp2_session_want_read'
/usr/bin/ld: libwazuhext.so: undefined reference to `nghttp2_session_del'
/usr/bin/ld: libwazuhext.so: undefined reference to `RTMP_Free'
/usr/bin/ld: libwazuhext.so: undefined reference to `nghttp2_pack_settings_payload'
/usr/bin/ld: libwazuhext.so: undefined reference to `nghttp2_strerror'
/usr/bin/ld: libwazuhext.so: undefined reference to `nghttp2_session_get_stream_user_data'
/usr/bin/ld: libwazuhext.so: undefined reference to `BrotliDecoderDecompressStream'
/usr/bin/ld: libwazuhext.so: undefined reference to `psl_is_cookie_domain_acceptable'
/usr/bin/ld: libwazuhext.so: undefined reference to `nghttp2_session_callbacks_set_send_callback'
/usr/bin/ld: libwazuhext.so: undefined reference to `nghttp2_session_callbacks_set_before_frame_send_callback'
/usr/bin/ld: libwazuhext.so: undefined reference to `RTMP_Write'
/usr/bin/ld: libwazuhext.so: undefined reference to `nghttp2_session_callbacks_del'
/usr/bin/ld: libwazuhext.so: undefined reference to `nghttp2_session_callbacks_set_on_begin_headers_callback'
/usr/bin/ld: libwazuhext.so: undefined reference to `nghttp2_session_callbacks_set_on_frame_recv_callback'
/usr/bin/ld: libwazuhext.so: undefined reference to `psl_get_version'
/usr/bin/ld: libwazuhext.so: undefined reference to `RTMP_Close'
/usr/bin/ld: libwazuhext.so: undefined reference to `nghttp2_session_send'
/usr/bin/ld: libwazuhext.so: undefined reference to `RTMP_Connect1'
/usr/bin/ld: libwazuhext.so: undefined reference to `nghttp2_session_callbacks_set_on_data_chunk_recv_callback'
/usr/bin/ld: libwazuhext.so: undefined reference to `nghttp2_session_callbacks_set_on_frame_not_send_callback'
/usr/bin/ld: libwazuhext.so: undefined reference to `BrotliDecoderVersion'
/usr/bin/ld: libwazuhext.so: undefined reference to `nghttp2_submit_settings'
collect2: error: ld returned 1 exit status
make[1]: *** [Makefile:857: wazuh-db] Error 1
analysisd/analysisd.c: In function ‘w_log_rotate_thread’:
analysisd/analysisd.c:2421:17: warning: ‘strncpy’ output may be truncated copying 3 bytes from a string of length 3 [-Wstringop-truncation]
                 strncpy(prev_month,mon, 3);
                 ^~~~~~~~~~~~~~~~~~~~~~~~~~
analysisd/analysisd.c: In function ‘OS_ReadMSG’:
analysisd/analysisd.c:805:9: warning: ‘strncpy’ output may be truncated copying 3 bytes from a string of length 3 [-Wstringop-truncation]
         strncpy(lf->mon, prev_month, 3);
         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
make[1]: Leaving directory '/home/braulio/Documents/GitHub/test/src'
make: *** [Makefile:544: server] Error 2
```
The issue is caused by a problem with the `libcurl`.

Regards.

**Comment by sempervictus** [comment]
Arch is a widely used target platform, we deploy it for everything from GPU compute servers to tiny storage systems to end-user laptops. The ABI tracking _can_ get annoying, but it provides benefit as a precursor to problems that'll be seen in "more stable" distros in X days/weeks/months when they pull in the packages Arch has - sort of a canary in the coalmine for ABI problems.

**Comment by BraulioV** [comment]
Hi @sempervictus,

Sorry for the super-late response. The compilation of the agent and the manager is currently broken by default due to an error related to the OpenSSL. The OpenSSL's version used to build the `libwazuhext.so` there's a bug that avoids loading the `glob` module in the Perl scripts used to configure the OpenSSL before compiling it. You can find more info about this in issue https://github.com/wazuh/wazuh/issues/4054.

Right now, you can build the agent in ArchLinux following these steps:
1. Install all the build dependencies: `pacman -Syu gcc git make automake autoconf libtool libnghttp2 libpsl librtmp0 brotli python perl-text-glob`.
2. Clone the repository: `git clone https://github.com/wazuh/wazuh --depth=1 -b v3.10.2`
3. Download the external dependencies: `cd wazuh && make -C src deps`
4. Apply the following patch to build the OpenSSL:
```
sed -i "s#'File::Glob' => qw/glob/;#'File::Glob' => qw/bsd_glob/;#g" src/external/openssl/test/recipes/90-test_fuzz.t
sed -i "s#'File::Glob' => qw/glob/;#'File::Glob' => qw/bsd_glob/;#g" src/external/openssl/test/recipes/80-test_ssl_new.t
sed -i "s#'File::Glob' => qw/glob/;#'File::Glob' => qw/bsd_glob/;#g" src/external/openssl/test/recipes/40-test_rehash.t
sed -i "s#'File::Glob' => qw/glob/;#'File::Glob' => qw/bsd_glob/;#g" src/external/openssl/test/build.info
sed -i "s#'File::Glob' => qw/glob/;#'File::Glob' => qw/bsd_glob/;#g" src/external/openssl/test/run_tests.pl
sed -i "s#'File::Glob' => qw/glob/;#'File::Glob' => qw/bsd_glob/;#g" src/external/openssl/util/process_docs.pl
sed -i "s#'File::Glob' => qw/glob/;#'File::Glob' => qw/bsd_glob/;#g" src/external/openssl/Configure
```
5. Build the agent: `make -C src TARGET=agent`.
6. Run the `install.sh` to install the agent.

To build the manager, follow these steps:
1. Install the build dependencies: `pacman -Syu gcc git make automake autoconf libtool libnghttp2 libpsl librtmp0 brotli python perl-text-glob expat bzip2 gdbm openssl libffi zlib libnsl tk sqlite valgrind bluez-libs mpdecimal llvm gdb xorg-server-xvfb`
2. Clone the repository: `git clone https://github.com/wazuh/wazuh --depth=1 -b v3.10.2`
3. Download the external dependencies: `cd wazuh && make -C src deps`
4. Patch the OpenSSL:
```
sed -i "s#'File::Glob' => qw/glob/;#'File::Glob' => qw/bsd_glob/;#g" src/external/openssl/test/recipes/90-test_fuzz.t
sed -i "s#'File::Glob' => qw/glob/;#'File::Glob' => qw/bsd_glob/;#g" src/external/openssl/test/recipes/80-test_ssl_new.t
sed -i "s#'File::Glob' => qw/glob/;#'File::Glob' => qw/bsd_glob/;#g" src/external/openssl/test/recipes/40-test_rehash.t
sed -i "s#'File::Glob' => qw/glob/;#'File::Glob' => qw/bsd_glob/;#g" src/external/openssl/test/build.info
sed -i "s#'File::Glob' => qw/glob/;#'File::Glob' => qw/bsd_glob/;#g" src/external/openssl/test/run_tests.pl
sed -i "s#'File::Glob' => qw/glob/;#'File::Glob' => qw/bsd_glob/;#g" src/external/openssl/util/process_docs.pl
sed -i "s#'File::Glob' => qw/glob/;#'File::Glob' => qw/bsd_glob/;#g" src/external/openssl/Configure
```
5. Build the manager: `LIBS="-lnghttp2 -lbrotlidec -lpsl" make -C src TARGET=server -j $(nproc)`
6. Install the manager by running: `./install.sh`.

I hope this helps and sorry for the inconveniences.

**Comment by vikman90** [comment]
@BraulioV @sempervictus this issue seems to be already fixed as #4054 is closed.

Let me close this issue. Should this problem persist in your environment, please reopen it.
Best regards.
